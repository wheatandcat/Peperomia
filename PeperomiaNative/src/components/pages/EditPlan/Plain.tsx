import React, { useState, memo, useCallback } from 'react';
import { SuggestItem } from 'lib/suggest';
import { useDidMount } from 'hooks/index';
import { ContextProps as ItemContextProps } from 'containers/Items';
import Page from 'components/templates/CreatePlan/Page';
import { Props } from './';

type PlainProps = Props &
  Pick<ItemContextProps, 'items' | 'refreshData'> & {
    input: {
      title: string;
      date: string;
    };
    kind: string;
    onInput: (name: string, value: any) => void;
    onSave: () => void;
    onIcons: () => void;
    onHome: () => void;
  };

type PlainState = {
  suggestList: SuggestItem[];
};

const Plain = memo((props: PlainProps) => {
  const [state, setState] = useState<PlainState>({
    suggestList: [],
  });

  useDidMount(() => {
    const suggestList = (props.items || []).map((item) => ({
      title: item.title,
      kind: item.kind,
    }));

    setState((s) => ({
      ...s,
      suggestList,
    }));
  });

  const onSave = useCallback(async () => {
    await props.onSave();

    if (props.refreshData) {
      props.refreshData();
    }
  }, [props]);

  return (
    <Page
      mode="edit"
      title={props.input.title}
      date={props.input.date}
      kind={props.kind}
      suggestList={state.suggestList}
      onInput={props.onInput}
      onSave={onSave}
      onIcons={props.onIcons}
      onHome={props.onHome}
    />
  );
});

export default Plain;
