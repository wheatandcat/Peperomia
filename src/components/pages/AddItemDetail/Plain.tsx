import React from 'react';
import { CreateItemDetailMutationResult } from 'queries/api/index';
import { ConnectedType } from './Connected';
import Page from './Page';

export type QueryProps = {
  mutationData: CreateItemDetailMutationResult;
};

type Props = QueryProps & ConnectedType;

const Plain: React.FC<Props> = (props) => {
  return <Page {...props} onDismiss={props.onDismiss} onSave={props.onSave} />;
};

export default Plain;
