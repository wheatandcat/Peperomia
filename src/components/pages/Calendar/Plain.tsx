import alertDialog from 'lib/alertDialog';
import ErrorPage from 'components/organisms/Error/Error';
import React from 'react';
import { CalendarQueryResult } from 'queries/api/index';
import Loading from 'components/atoms/Loading';
import { ConnectedType } from './Connected';
import Page from './Page';

export type QueryProps = Pick<
  CalendarQueryResult,
  'data' | 'loading' | 'error'
>;

type Props = QueryProps & ConnectedType;

const Plain: React.FC<Props> = (props) => {
  if (alertDialog(props)) return <ErrorPage {...props} />;
  if (props.loading) return <Loading loading />;

  const calendar = props.data?.calendar!;

  console.log(calendar);

  return <Page calendar={calendar} onDismiss={props.onDismiss} />;
};

export default Plain;
