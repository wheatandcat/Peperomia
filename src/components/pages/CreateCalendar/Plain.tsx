import React from 'react';
import { CreateCalendarMutationResult } from 'queries/api/index';
import { ConnectedType } from './Connected';
import Page from './Page';

export type QueryProps = {
  mutationData: CreateCalendarMutationResult;
};

type Props = QueryProps & ConnectedType;

const Plain: React.FC<Props> = (props) => {
  return <Page {...props} onDismiss={props.onDismiss} onSave={props.onSave} />;
};

export default Plain;
