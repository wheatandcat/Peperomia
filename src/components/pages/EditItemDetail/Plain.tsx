import React from 'react';
import {
  UpdateItemDetailMutationResult,
  ItemDetailQueryHookResult,
} from 'queries/api/index';
import Loading from 'components/atoms/Loading';
import alertDialog from 'lib/alertDialog';
import ErrorPage from 'components/organisms/Error/Error';
import { ConnectedType } from './Connected';
import Page from './Page';

export type QueryProps = {
  mutationData: UpdateItemDetailMutationResult;
} & Pick<ItemDetailQueryHookResult, 'data' | 'loading' | 'error'>;

type Props = QueryProps & ConnectedType;

const Plain: React.FC<Props> = (props) => {
  if (alertDialog(props)) return <ErrorPage {...props} />;
  if (props.loading) return <Loading loading />;

  const itemDetail = props.data?.itemDetail!;

  return <Page {...props} itemDetail={itemDetail} />;
};

export default Plain;
