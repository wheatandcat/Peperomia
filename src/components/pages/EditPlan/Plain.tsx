import React, { memo, useCallback } from 'react';
import { ContextProps as ItemContextProps } from 'containers/Items';
import Page from 'components/templates/CreatePlan/Page';
import { ConnectType } from './Connected';
import { Props as IndexProps } from './';

type Props = IndexProps &
  ConnectType &
  Pick<ItemContextProps, 'items' | 'refreshData'>;

const Plain: React.FC<Props> = (props) => {
  const onSave = useCallback(async () => {
    await props.onSave();

    if (props.refreshData) {
      props.refreshData();
    }
  }, [props]);

  return (
    <Page
      mode="edit"
      title={props.input.title}
      date={props.input.date}
      kind={props.kind}
      onInput={props.onInput}
      onSave={onSave}
      onIcons={props.onIcons}
      onHome={props.onHome}
      showActionSheetWithOptions={props.showActionSheetWithOptions}
    />
  );
};

export default memo(Plain);
