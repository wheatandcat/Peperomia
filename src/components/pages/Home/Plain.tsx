import React, { memo, useState, useCallback, useEffect } from 'react';
import { useNavigation } from '@react-navigation/native';
import { Alert } from 'react-native';
import uuidv1 from 'uuid/v1';
import { deleteItem } from 'lib/item';
import { ContextProps as ItemContextProps } from 'containers/Items';
import { ContextProps as AuthContextProps } from 'containers/Auth';
import theme, { darkMode } from 'config/theme';
import { SelectItem } from 'domain/item';
import FocusAwareStatusBar from 'components/organisms/FocusAwareStatusBar';
import Page from './Page';

export type ConnectType = {
  onSchedule: (itemId: string, title: string) => void;
  onDelete: (itemId: string) => void;
};

export type ItemProps = SelectItem & {
  id: string;
  about: string;
};

export type Props = Pick<ItemContextProps, 'items' | 'about' | 'refreshData'> &
  Pick<AuthContextProps, 'uid'> & {
    loading: boolean;
    refresh: string;
  };

type State = {
  refresh: string;
};

const HomePlain: React.FC<Props> = (props) => {
  const { navigate } = useNavigation();
  const [state, setState] = useState<State>({ refresh: '' });
  const uid = props.uid;

  useEffect(() => {
    if (state.refresh === props.refresh) {
      return;
    }
    setState({ refresh: props.refresh });

    if (props.refreshData) {
      props.refreshData();
    }
  }, [props, props.refresh, state.refresh]);

  const onDelete = useCallback(
    async (itemId: string) => {
      const ok = await deleteItem(uid, { id: itemId });
      if (!ok) {
        Alert.alert('削除に失敗しました');
        return;
      }

      setState({ refresh: uuidv1() });
    },
    [uid]
  );

  const onSchedule = useCallback(
    (id: string, title: string) => {
      navigate('Schedule', { itemId: id, title });
    },
    [navigate]
  );

  const items = (props.items || []).map((item) => {
    const about = (props.about || []).find((val) => val.itemId === item.id);

    return {
      ...item,
      id: String(item.id),
      about: about ? about.about : '',
    };
  });

  return (
    <>
      <FocusAwareStatusBar
        backgroundColor={darkMode() ? theme().color.black : theme().color.main}
        barStyle={darkMode() ? 'light-content' : 'dark-content'}
      />
      <Page
        data={items}
        loading={props.loading}
        onSchedule={onSchedule}
        onDelete={onDelete}
      />
    </>
  );
};

export default memo(HomePlain);
