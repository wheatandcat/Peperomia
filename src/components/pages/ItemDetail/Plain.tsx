import alertDialog from 'lib/alertDialog';
import ErrorPage from 'components/organisms/Error/Error';
import React, { memo } from 'react';
import { ItemDetailQueryHookResult } from 'queries/api/index';
import Loading from 'components/atoms/Loading';
import { ConnectedType } from './Connected';
import Page from './Page';

export type QueryProps = Pick<
  ItemDetailQueryHookResult,
  'data' | 'loading' | 'error'
>;

type Props = QueryProps & ConnectedType;

const Plain: React.FC<Props> = (props) => {
  if (alertDialog(props)) return <ErrorPage {...props} />;
  if (props.loading) return <Loading loading />;

  const itemDetail = props.data?.itemDetail!;

  return (
    <Page
      itemDetail={itemDetail}
      date={props.date}
      onDismiss={props.onDismiss}
    />
  );
};

export default memo(Plain);
