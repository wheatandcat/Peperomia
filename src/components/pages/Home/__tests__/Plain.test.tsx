import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Plain from '../Plain';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useFocusEffect: () => jest.fn(),
    useNavigation: () => ({
      navigation: () => jest.fn(),
    }),
  };
});

describe('components/pages/Home/Plain.tsx', () => {
  let wrapper: ShallowWrapper;

  const propsData: any = () => ({
    refreshData: jest.fn(),
    items: [],
    about: '',
    refresh: false,
    loading: false,
    uid: 'test',
  });

  beforeEach(() => {
    wrapper = shallow(<Plain {...propsData()} />);
  });

  it('正常に表示されている', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
