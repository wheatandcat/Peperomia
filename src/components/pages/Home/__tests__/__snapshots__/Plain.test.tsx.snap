// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Home/Plain.tsx 正常に表示されている 1`] = `
<Fragment>
  <FocusAwareStatusBar
    backgroundColor="#006835"
    barStyle="dark-content"
  />
  <HomePage
    data={Array []}
    loading={false}
    onDelete={[Function]}
    onSchedule={[Function]}
  />
</Fragment>
`;
